parser.y

```diff
@@ -202,6 +203,73 @@ rule
 
   token_declaration: id int_opt alias { result = val }
 
+  rule_declaration: "%rule" IDENTIFIER "(" rule_args ")" ":" rule_rhs_list
+                      {
+                        builder = Grammar::ParameterizingRuleBuilder.new(val[1].s_value, val[3], val[6])
+                        @grammar.add_parameterizing_rule_builder(builder)
+                      }
+
+  rule_args: IDENTIFIER { result = [val[0]] }
+           | rule_args "," IDENTIFIER { result = val[0].append(val[2]) }
+
+  rule_rhs_list: rule_rhs
+                {
+                  builder = val[0]
+                  result = [builder]
+                }
+          | rule_rhs_list "|" rule_rhs
+                {
+                  builder = val[2]
+                  result = val[0].append(builder)
+                }
+
+  rule_rhs: /* empty */
```

Comment:
[Note] Rule for nested parameterizing rules like `rhs IDENTIFIER parameterizing_suffix tag_opt` or `rhs IDENTIFIER "(" parameterizing_args ")" tag_opt` are not defined now. It will be supported by another PR.